### PR TITLE
platform/mikro-e: enable slip protocol

### PR DESCRIPTION
**Pull request by manohar-narkhede-imgtec**
Signed-off-by: Manohar Narkhede <Manohar.Narkhede@imgtec.com>

### CODE REVIEW

File: platform/mikro-e/Makefile.mikro-e [hunk 1]
```diff
@@ -42,7 +46,7 @@ endif
 
 MODULES += core/net core/net/mac core/net/llsec
 
-CONTIKI_SOURCEFILES += $(CONTIKI_TARGET_SOURCEFILES)
+CONTIKI_SOURCEFILES += $(CONTIKI_TARGET_SOURCEFILES) slip.c
```
Comment (Avinash95) [inline platform/mikro-e/Makefile.mikro-e]: you can append the slip.c filename to CONTIKI_TARGET_SOURCEFILES variable as it is defined in the same makefile

Comment (manohar-narkhede-imgtec) [reply platform/mikro-e/Makefile.mikro-e]: It is not TARGET(Mikro-e) related.

**Review by mtusnio — CHANGES_REQUESTED** [review]
Does RPL work without SLIP? What's the reason we need SLIP configured by default? I'm a bit worried here since this would use one UART line for debug, one for SLIP, leaving no room for clickers using UART.

File: platform/mikro-e/Makefile.mikro-e (hunk 1 above)
Comment (Avinash95) [reply platform/mikro-e/Makefile.mikro-e]: But it's inside target(Mikro-e) makefile, so isn't it related to Mikro-e?

Comment (manohar-narkhede-imgtec) [reply platform/mikro-e/Makefile.mikro-e]: slip.c is Contiki common source code so it should be part of CONTIKI_SOURCEFILES which is independent of the platform. The platform specific source code will be part of CONTIKI_TARGET_SOURCEFILES.

**Comment by manohar-narkhede-imgtec** [comment]
RPL works without SLIP. RPL as such does not have any dependency on SLIP. 

As the rpl border router was running on Clicker, the SLIP required for its functionality. The border router can run on Ci 40.

**Comment by mtusnio** [comment]
I don't like adding SLIP by default, how do you configure defines for a border router? It'd make sense to add SLIP then rather than have it all the time. What's exactly SLIP used for in RPL, what does it communicate with, or is it just for debug?

**Comment by Shpinkso** [comment]
What's happening with this please, don't just let it sit here forever.

**Comment by manohar-narkhede-imgtec** [comment]
SLIP has nothing to do with. It is implementation of Serial Line Internet Protocol which provides outside n/w connectibity.

The border router can be used without RPL. It is part of core Contiki source code. Already there is source code getting compiled even if that is not getting used by application. Should we remove that code also from Makefile? The examples are common for other platforms also.

**Comment by mtusnio** [comment]
We do not enable any other UART apart from the one we use for debug by default, @francois-berder-imgtec you mentioned you will do a PR for that, is this one blocked until that's done?

**Comment by mtusnio** [comment]
I believe we need https://github.com/CreatorDev/contiki/pull/69 merged first.

**Comment by mtusnio** [comment]
@manohar-narkhede-imgtec Can you rebase this branch and modify the PR to match changes done by @francois-berder-imgtec ?

File: platform/mikro-e/Makefile.mikro-e [hunk 2]
```diff
@@ -15,9 +15,17 @@ CONTIKI_PLAT_DEFS += -D __USE_UART_PORT3__
 ifeq ($(USE_SERIAL_PADS),1)
   #If using RB9 and RD11 as uart pins for debug
   CONTIKI_PLAT_DEFS += -D __USE_UART_PORT2_FOR_DEBUG__
+ifeq ($(USE_SLIP),1)
+  CONTIKI_PLAT_DEFS += -D __USE_UART_PORT3__
```
Comment (mtusnio) [inline platform/mikro-e/Makefile.mikro-e]: Francois's change already added this by default, we can get rid of the extra -D __USE_UART_PORT3__

File: platform/mikro-e/Makefile.mikro-e [hunk 3]
```diff
@@ -15,9 +15,17 @@ CONTIKI_PLAT_DEFS += -D __USE_UART_PORT3__
...
+  CONTIKI_PLAT_DEFS += -D __USE_UART_PORT3_FOR_SLIP__
+endif
 else
   #If using RF4 and RF5 as uart pins for debug
   CONTIKI_PLAT_DEFS += -D __USE_UART_PORT3_FOR_DEBUG__
+ifeq ($(USE_SLIP),1)
+  CONTIKI_PLAT_DEFS += -D __USE_UART_PORT2__
```
Comment (mtusnio) [inline platform/mikro-e/Makefile.mikro-e]: Can remove this one as well

**Review by mtusnio — CHANGES_REQUESTED** [review]

**Pull request closed without merging**